pykafka/simpleconsumer.py

```diff
@@ -507,10 +507,13 @@ def _handle_success(parts):
 
         log.info("Resetting offsets for %s partitions", len(list(owned_partition_offsets)))
 
+        # sort offsets to avoid deadlocks
+        sorted_offsets = sorted(iteritems(owned_partition_offsets), key=lambda k: k[0].partition.id)
```

Comment:
I think you may want to recompute this within your retry loop since owned_partition_offsets is mutated by the loop.